hypervisor/arch/x86/configs/pci_dev.c

```diff
@@ -28,25 +22,19 @@ static bool is_allocated_to_prelaunched_vm(struct pci_pdev *pdev)
 	struct acrn_vm_config *vm_config;
 	struct acrn_vm_pci_dev_config *dev_config;
 
-	for (vmid = 0U; vmid < CONFIG_MAX_VM_NUM; vmid++) {
+	for (vmid = 0U; (vmid < CONFIG_MAX_VM_NUM) && !found; vmid++) {
 		vm_config = get_vm_config(vmid);
-		if (vm_config->load_order != PRE_LAUNCHED_VM) {
-			continue;
-		}
-
-		for (pci_idx = 0U; pci_idx < vm_config->pci_dev_num; pci_idx++) {
-			dev_config = &vm_config->pci_devs[pci_idx];
-			if ((dev_config->emu_type == PCI_DEV_TYPE_PTDEV) &&
-					bdf_is_equal(dev_config->pbdf, pdev->bdf)) {
-				dev_config->pdev = pdev;
-				found = true;
-				break;
+		if (vm_config->load_order == PRE_LAUNCHED_VM) {
+			for (pci_idx = 0U; pci_idx < vm_config->pci_dev_num; pci_idx++) {
```

Comment:
WHy not use !found too?

Reply:
If we have found, we hope to break the outside for loop. So if we came here, we still not found the device in pre-launched VM.